web/pages/review/cards.tsx

```diff
@@ -0,0 +1,107 @@
+import { getRequest } from "@src/apiRequest";
+import ProtectedRoute from "@components/ProtectedRoute";
+import Flip from "@components/routeReview/Flip";
+import { NextFlashcardGetReq } from "@src/types/NextFlashcardGetReq";
+import { NextFlashcardGetResp } from "@src/types/NextFlashcardGetResp";
+import { FlashcardFromIDResp } from "@src/types/FlashcardFromIDResp";
+import { FlashcardFromIDGet } from "@src/types/FlashcardFromIDGet";
+import Button from "@components/Button";
+/*
+This page is accessed when the user has set modes and flashcards in localStorage. 
+This page sends a request to /review/next to get the next 'packet' of
+upcomingReview info. Then, it requests the flashcard info from ID via 
+/flashcard/fromid and displays this.
+The actual interfaces for the revision are in /src/components/routeReview
+*/
+
+export default function ReviewCards() {
+    const auth = useAuth();
+    const [reviewModes] = useSessionStorage(reviewModesKey, "");
+    const [reviewFIDs] = useSessionStorage(reviewFidsKey, "");
+
+    const { data: upcomingReviews } = useQuery({
+        // gets the next packet of review modes and flashcards
+        queryKey: ["upcomingReviews", reviewModes, reviewFIDs, auth.user],
+        queryFn: async () => {
+            if (!auth.user?.id_token) {
+                return Promise.reject("Unauthorised");
+            }
+            const resp = await getRequest({
+                path: "/review/next",
+                id_token: auth.user.id_token,
+                queryParams: {
+                    modes: reviewModes,
+                    flashcardIds: reviewFIDs,
+                } as NextFlashcardGetReq,
+            });
+            if (resp.status != 200) {
+                return Promise.reject(resp.body);
+            } else {
+                return (await resp.json()) as NextFlashcardGetResp;
+            }
+        },
+        enabled: !!(reviewModes && reviewFIDs && auth.user?.id_token),
+    });
+
+    const { data: flashcard } = useQuery({
+        // gets the flashcard info from id (SINGULAR)
+        queryKey: ["get-flashcards", reviewFIDs],
+        queryFn: async () => {
+            if (!auth.user?.id_token) {
+                return Promise.reject(
+                    "tanstack react-query enabled bypassed: Unauthorised error"
+                );
+            }
+            if (!upcomingReviews) {
+                return Promise.reject(
+                    "tanstack react-query enabled bypassed: No upcoming reviews"
+                );
+            }
+            if (upcomingReviews.flashcardIds.split(",").length > 1) {
+                return Promise.reject("MATCH not supported");
+            }
+
+            const resp = await getRequest({
+                path: "/flashcard/fromid",
+                id_token: auth.user.id_token,
+                queryParams: {
+                    modes: reviewModes.split(","),
+                    flashcardId: parseInt(upcomingReviews.flashcardIds),
```

Comment:
Will this have to be changed to support multiple (like having a bulk flashcard from id endpoint)?

Reply:
Yes, we'd have to rewrite as a Promise.all like what you mentioned in another comment for getting the accessed flashcards from folders. This would surely end up being quite inefficient, requesting large amounts of individual flashcards from their ID. 
I suppose for each next immediate review page, it would only either be 1 request (flip, write) or ~10 (match), but there will be a better way to do this, like you said, as a bulk fromid endpoint, or perhaps something like caching terms and definitions of accessed cards when they're selected for review. Your suggestion certainly seems simpler although I think it will be a duplication of requests as we get the term and definition during selection?
Either way, I haven't implemented any solution yet.